kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -0,0 +1,208 @@
+                "%MESSAGE_COUNT%", "\"" + numOfMessages + "\"",
+                "%MESSAGE%", message));
+        kubeClient().getClient().load(new FileInputStream(file)).inNamespace(deployNamespace).create();
+        return getPodNameByLabel(deployNamespace, "app", Constants.KAFKA_PRODUCER_CLIENT_LABEL, Duration.ofSeconds(10).toMillis());
+    }
+
+    private static File replaceStringInResourceFile(String resourceTemplateFileName, String newResourceFileName, Map<String, String> replacements) throws IOException {
+        Path path = Path.of(Objects.requireNonNull(KafkaUtils.class
+                .getClassLoader().getResource(resourceTemplateFileName)).getPath());
+        Charset charset = StandardCharsets.UTF_8;
+
+        String content = new String(Files.readAllBytes(path), charset);
+        for (Map.Entry<String, String> entry : replacements.entrySet()) {
+            content = content.replaceAll(entry.getKey(), entry.getValue());
+        }
+        String resourceFile = "/tmp/" + newResourceFileName;
+
+        Files.writeString(Path.of(resourceFile), content, charset);
+        File file = new File(resourceFile);
+        file.deleteOnExit();
+        return file;
+    }
+
+    /**
+     * Delete all jobs.
+     *
+     * @param deployNamespace the deploy namespace
+     */
+    public static void deleteAllJobs(String deployNamespace) {
+        LOGGER.info("Deleting producer and consumer jobs in {} namespace", deployNamespace);
+        kubeClient().getClient().batch().v1().jobs().inNamespace(deployNamespace).delete();
+        kubeClient().getClient().pods().inNamespace(deployNamespace).withLabel("app", Constants.KAFKA_PRODUCER_CLIENT_LABEL).delete();
+        kubeClient().getClient().pods().inNamespace(deployNamespace).withLabel("app", Constants.KAFKA_CONSUMER_CLIENT_LABEL).delete();
+        kubeClient().getClient().pods().inNamespace(deployNamespace).withLabel("app", Constants.KAFKA_CONSUMER_CLIENT_LABEL).waitUntilCondition(x -> x == null, 10,
+                TimeUnit.SECONDS);
+    }
+
+    /**
+     * Restart broker boolean.
+     *
+     * @param deployNamespace the deploy namespace
+     * @param clusterName the cluster name
+     * @return the boolean
+     */
+    public static boolean restartBroker(String deployNamespace, String clusterName) {
+        String podName = "";
+        List<Pod> kafkaPods = kubeClient().getClient().pods().inNamespace(Constants.KROXY_DEFAULT_NAMESPACE).list().getItems();
+        int numOfBrokers = kafkaPods.size();
+        for (Pod pod : kafkaPods) {
+            String tmpName = pod.getMetadata().getName();
+            if (tmpName.startsWith(clusterName)) {
+                podName = pod.getMetadata().getName();
+            }
+        }
+        kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).delete();
+        kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).waitUntilCondition(Objects::isNull, 10, TimeUnit.SECONDS);
```

Comment:
this looks racey.  for a reliable test you'd need to await a the appearance of a new pod object with a new [uid]( https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids)